etcdmain/gateway.go

```diff
@@ -77,6 +78,23 @@ func newGatewayStartCommand() *cobra.Command {
 	return &cmd
 }
 
+func stripSchema(eps []string) []string {
```

Comment:
the discovery stuff should eventually be covered by e2e tests; I don't think a unit test is necessary here